Description: don't declare a Wants/Requires on network.target in systemd units
 Don't declare a Wants/Requires on network.target; this is a passive target
 that should only be pulled in by implementors of the networking service.
 The requirement for network needs to be expressed as a dependency on
 network-online.target.
Author: Steve Langasek <steve.langasek@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1440180

--- cloud-init-0.7.7~bzr1088.orig/systemd/cloud-config.service
+++ cloud-init-0.7.7~bzr1088/systemd/cloud-config.service
@@ -1,7 +1,7 @@
 [Unit]
 Description=Apply the settings specified in cloud-config
-After=network.target cloud-config.target syslog.target
-Wants=network.target cloud-config.target
+After=network-online.target cloud-config.target syslog.target
+Wants=network-online.target cloud-config.target
 
 [Service]
 Type=oneshot
--- cloud-init-0.7.7~bzr1088.orig/systemd/cloud-final.service
+++ cloud-init-0.7.7~bzr1088/systemd/cloud-final.service
@@ -1,7 +1,7 @@
 [Unit]
 Description=Execute cloud user/final scripts
-After=network.target cloud-config.service syslog.target rc-local.service
-Wants=network.target cloud-config.service
+After=network-online.target cloud-config.service syslog.target rc-local.service
+Wants=network-online.target cloud-config.service
 
 [Service]
 Type=oneshot
--- cloud-init-0.7.7~bzr1088.orig/systemd/cloud-init.service
+++ cloud-init-0.7.7~bzr1088/systemd/cloud-init.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Initial cloud-init job (metadata service crawler)
-After=local-fs.target network.target cloud-init-local.service
+After=local-fs.target network-online.target cloud-init-local.service
 Before=sshd.service sshd-keygen.service systemd-user-sessions.service
-Requires=network.target
+Requires=network-online.target
 Wants=local-fs.target cloud-init-local.service sshd.service sshd-keygen.service
 
 [Service]
